### PR TITLE
Updated Bad diagnostic errors

### LINKED ISSUE #64903: Bad diagnostic when inheriting from class in extension
The following errors are misleading and could use a rewording as clear language rules, depending on the kinds of extended and inherited types.

```swift
protocol P {}
struct S {}
class C1 {}
class C2 {}

extension C2: S {} // error: inheritance from non-protocol type 'S'
extension C2: C1 {} // error: inheritance from non-protocol type 'C1'
extension C2: C1 & P {} // error: inheritance from class-constrained protocol composition type 'C1 & P'
```

**Environment**
- Swift version 5.9-dev (LLVM 6a04c4848990c70, Swift c1d5118c21d)

NB: How to improve on these is left as an exercise to whoever takes the issue.

### PR DESCRIPTION
**Pull request by dk-talks**
Fixes #64903 

I have updated the error messages which were not very clear to the user.

Environment
Swift version 5.9-dev (LLVM 6a04c4848990c70, Swift f2bac1a22939a4d)

This is my first pull request on this project. Please scold me if I have made any mistake.

### CODE REVIEW

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -3056,7 +3056,7 @@ ERROR(duplicate_inheritance,none,
 WARNING(duplicate_anyobject_class_inheritance,none,
   "redundant inheritance from 'AnyObject' and Swift 3 'class' keyword", ())
 ERROR(inheritance_from_protocol_with_superclass,none,
-  "inheritance from class-constrained protocol composition type %0", (Type))
+  "Cannot inherit from class-constrained protocol composition type %0", (Type))
```
Comment (LucianoPAlmeida) [inline include/swift/AST/DiagnosticsSema.def:3059]: Nit: all diagnostics start with lower case
```suggestion
  "cannot inherit from class-constrained protocol composition type %0", (Type))
```

File: test/decl/enum/enumtest.swift [hunk 2]
```diff
@@ -446,7 +446,7 @@ class C {}
 protocol P {}
 
 enum E : C & P {}
-// expected-error@-1 {{inheritance from class-constrained protocol composition type 'C & P'}}
+// expected-error@-1 {{Cannot inherit from class-constrained protocol composition type 'C & P'}}
```
Comment (LucianoPAlmeida) [inline test/decl/enum/enumtest.swift:449]: Nit: Lower cased as well. For all the occurrences =]
```suggestion
// expected-error@-1 {{cannot inherit from class-constrained protocol composition type 'C & P'}}
```

**Comment by dk-talks** [comment]
Hey @LucianoPAlmeida I have committed the new changes you suggested, please have a look

File: test/Generics/inheritance.swift [hunk 3]
```diff
@@ -55,7 +55,7 @@ func testGenericInherit() {
 }
 
 
-struct SS<T> : T { } // expected-error{{inheritance from non-protocol type 'T'}}
+struct SS<T> : T { } // expected-error{{class can only inherit from protocol in extension declaration 'T'}}
```
Comment (LucianoPAlmeida) [inline test/Generics/inheritance.swift:58]: This is incorrect because declaration is a `struct` but diagnostic says is a `class`.

Comment (dk-talks) [reply test/Generics/inheritance.swift]: Can you have a look now?

**Comment by LucianoPAlmeida** [comment]
Sorry, been a bit busy lately with work so I just glanced on the basics when commenting.
Now was trying to get more context I see that the exercise here is to try to make the wording of diagnostic to be clear on what the language rule is and tell that to the user. 
Here are a few things to think about:
```swift
struct SS<T> : T { } // expected-error{{cannot inherit from non-protocol type 'T'}}
```
It is still not clear what cannot inherit from not protocol types. So you can start thinking about the rules about inheritance for struct types and how to describe that in a succinct, but clear way. 

Same applies to other cases, so you can start with questions such as:
What is the language rules for inheritance of each construct `class`, `struct`, `enum` and extensions of each of those?
And how improve the current diagnostics to express that in a more clear way?

**Pull request closed without merging**